qiskit/test/mock/fake_backend.py

```diff
@@ -38,6 +62,257 @@ def __init__(self, token="123456", url="https://"):
+
+    def run(self, run_input, **kwargs):
+        """Main job in simulator"""
+        circuits = run_input
+        pulse_job = None
+        if isinstance(circuits, (pulse.Schedule, pulse.ScheduleBlock)):
+            pulse_job = True
+        elif isinstance(circuits, circuit.QuantumCircuit):
+            pulse_job = False
+        elif isinstance(circuits, list):
+            if circuits:
+                if all(isinstance(x, (pulse.Schedule, pulse.ScheduleBlock)) for x in circuits):
+                    pulse_job = True
+                elif all(isinstance(x, circuit.QuantumCircuit) for x in circuits):
+                    pulse_job = False
+        if pulse_job is None:
+            raise QiskitError(
+                "Invalid input object %s, must be either a "
+                "QuantumCircuit, Schedule, or a list of either" % circuits
+            )
+        if _optionals.HAS_AER:
+            from qiskit.providers import aer
+
+            if pulse_job:
+                from qiskit.providers.aer.pulse import PulseSystemModel
+
+                system_model = PulseSystemModel.from_backend(self)
+                sim = aer.Aer.get_backend("pulse_simulator")
+                job = sim.run(circuits, system_model=system_model, **kwargs)
+            else:
+                sim = aer.Aer.get_backend("qasm_simulator")
+                if self._props_dict:
+                    noise_model = self._get_noise_model_from_backend_v2(self, warnings=False)
+                    job = sim.run(circuits, noise_model=noise_model, **kwargs)
+                else:
+                    job = sim.run(circuits, **kwargs)
+        else:
+            if pulse_job:
+                raise QiskitError("Unable to run pulse schedules without qiskit-aer installed")
+            warnings.warn("Aer not found using BasicAer and no noise", RuntimeWarning)
+            sim = basicaer.BasicAer.get_backend("qasm_simulator")
+            job = sim.run(circuits, **kwargs)
+        return job
+
+    def _get_noise_model_from_backend_v2(
+        self,
+        gate_error=True,
+        readout_error=True,
+        thermal_relaxation=True,
+        temperature=0,
+        gate_lengths=None,
+        gate_length_units="ns",
+        standard_gates=None,
+        warnings=True,
+    ):
+        """Build noise model from BackendV2.
+        This is a temporary fix until Aer supports V2 backends.
+        """
+
+        from warnings import warn, catch_warnings, filterwarnings
+        from qiskit.circuit import Delay
+        from qiskit.providers.exceptions import BackendPropertyError
+        from qiskit.providers.aer.noise import NoiseModel
+        from qiskit.providers.aer.noise.device.models import (
+            _excited_population,
+            basic_device_gate_errors,
+            basic_device_readout_errors,
+        )
+        from qiskit.providers.aer.noise.passes import RelaxationNoisePass
+
+        properties = BackendProperties.from_dict(self._props_dict)
```

Comment:
While this works for this case because we have properties available, to fix this in https://github.com/Qiskit/qiskit-aer/issues/1477 we'll need to do this directly via the qubit properties and or target

Reply:
Yeah I did it this way because I don't want to copy more code from Aer to Terra. In order to support building noise model from qubit properties and / target, we need to modify [`basic_device_readout_errors`](https://github.com/Qiskit/qiskit-aer/blob/71d55588a338c02740e857fceef8439412746e20/qiskit/providers/aer/noise/device/models.py#L36) and [`basic_device_gate_errors`](https://github.com/Qiskit/qiskit-aer/blob/71d55588a338c02740e857fceef8439412746e20/qiskit/providers/aer/noise/device/models.py#L55) in `aer.noise.device.models.py`.

Would you want me to do that? Or would you keep it as is in Terra and fix that in Aer https://github.com/Qiskit/qiskit-aer/issues/1477